tests/e2e/lib.sh

```diff
@@ -276,7 +276,18 @@ deploy_sensor_via_operator() {
         --output-secrets - \
     | kubectl -n stackrox apply -f -
 
-    kubectl apply -n stackrox -f tests/e2e/yaml/secured-cluster-cr.yaml
+
+    # COLLECTION_METHOD should already be set, but let's make 100% sure that it is
+    if [[ -n "$COLLECTION_METHOD:-}" ]]; then
+        echo "Using COLLECTION_METHOD=${COLLECTION_METHOD}"
+        template_cr_file=tests/e2e/yaml/secured-cluster-cr-template.yaml
+        gen_cr_file=tests/e2e/yaml/secured-cluster-cr-gen.yaml
+        upper_case_collection_method="$(echo "$COLLECTION_METHOD" | tr '[:lower:]' '[:upper:]')"
+        sed "s|__COLLECTION_METHOD__|$upper_case_collection_method|" "$template_cr_file" > "$gen_cr_file"
+        kubectl apply -n stackrox -f "$gen_cr_file"
+    else
+        kubectl apply -n stackrox -f tests/e2e/yaml/secured-cluster-cr.yaml
```

Comment:
I don't like this. Now we have `secured-cluster-cr-template.yaml` and `secured-cluster-cr.yaml`. They could diverge. I though about using yq to add the collection method to `secured-cluster-cr.yaml`, but I have had negative experiences with it recently.

Reply:
Take a look at `tests/e2e/yaml/central-cr.envsubst.yaml` and how it's dealt with in this same script.
I believe `COLLECTION_METHOD` should be set somewhere higher up the stack. If it's not set, it would be worth doing that explicitly.

Reply:
> I believe COLLECTION_METHOD should be set somewhere higher up the stack. If it's not set, it would be worth doing that explicitly.

COLLECTION_METHOD is set for all e2e tests (I think) to ebpf here: https://github.com/stackrox/stackrox/blob/a3b98c5fea418c7ffa1832798b3e705ff5551e08/tests/e2e/lib.sh#L115

I wonder if there is some benefit to removing this and have the operator install progress with no explicit collection set in that yaml?

Reply:
In my opinion, we get the most value from running Operator-based and non-Operator-based deployments in e2e test context with consistent setting in _default_ scenario and consistent setting in _other_ scenario.

_default_ value for Operator == _default_ value for non-operator
_other_ value for Operator == _other_ value for non-operator

I use _default_ and _other_ in the same meaning as [here](https://redhat-internal.slack.com/archives/CFMQ5C2TT/p1693390688428029?thread_ts=1693326299.856949&cid=CFMQ5C2TT). I.e. currently _default_==eBPF, _other_==CO-RE, but some time soon it's going to be flipped.

Whether we provide the _default_ value (e.g. eBPF) explicitly or rely on true defaulting, i.e. not provide anything and rely on the operator to do "the right thing", is a good question. I think @JoukoVirtanen and @stackrox/draco are better positioned to answer about how to better test the defaulting logic (especially since there might be considerations around how existing clusters are upgraded).
Since this is a difficult question which may not have a simple answer quickly, my suggestion would be to opt for providing explicit _default_ for now.